src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -384,59 +384,69 @@ def process_messages(
 
     org_strings = defaultdict(set)
     strings = set()
-    skipped_offsets = set()
+    skipped_offsets: Mapping[int, Set[int]] = defaultdict(set)
     with metrics.timer("process_messages.parse_outer_message"):
-        parsed_payloads_by_offset: MutableMapping[int, json.JSONData] = {}
+        parsed_payloads_by_offset: MutableMapping[
```

Comment:
you could make this type a `MutableMapping[Tuple[int, int], json.JSONData]`, then you don't need the nested for-loop. Same with `skipped_offsets`, it could be `Set[Tuple[int, int]]`, I would think this makes usage a little simpler.